api/internal.go

```diff
@@ -40,5 +40,5 @@ func (w Wrapper) TaskUpdate(ctx echo.Context, customerID int, taskID string) err
 		return err
 	}
 
-	return ctx.NoContent(http.StatusAccepted)
+	return ctx.JSON(http.StatusOK, task)
```

Comment:
This returns the task as provided by the client for the PUT operation, but not necessarily the task in the system, since only the `task.state` will be updated, other fields might be ignored. I think you should get it from the store and return the updated version.